dist/profile/templates/buildmaster/casc/clouds.yaml.erb

```diff
@@ -0,0 +1,171 @@
+---
+jenkins:
+  clouds:
+  <%- if @azure_vm_agents && !@azure_vm_agents.empty? -%>
+  - azureVM:
+      azureCredentialsId: "azure-credentials"
+      cloudName: "Azure"
+      configurationStatus: "pass"
+      deploymentTimeout: 1200
+      <%- $maxAzureVms = 0; @azure_vm_agents.each do |agent| -%>
+      <%- $maxAzureVms = agent["maxInstances"] + $maxAzureVms -%>
```

Comment:
not very obvious imo, it's just summing the max count across the clouds not a max on each template but I guess ok

Reply:
Yes, it's summing for all the agents of a given cloud, to ensure that the cloud total is the sum of all its templates. (The expressivity of this code is bad, but I lost half of my hair trying to do a simple "inline" instruction in ERB, not mentioning trying to use the "reduce" lamba function of Puppet) so I ended up typing this code angrily: it works but it's not good ©